File: cryptography-cipher/src/main/java/dev/retrotv/crypto/cipher/block/algorithm/LEA.java
package dev.retrotv.crypto.cipher.block.algorithm;

import dev.retrotv.crypto.cipher.block.BlockCipher;
import org.bouncycastle.crypto.engines.LEAEngine;

import static dev.retrotv.crypto.cipher.enums.ECipher.LEA;

/**
 * LEA 암호화 알고리즘을 사용하는 블록 암호화 클래스 입니다.
 */
public class LEA extends BlockCipher {
    public LEA() {
        this.engine = new LEAEngine();
        this.algorithm = LEA;
    }
}

File: cryptography-cipher/src/main/kotlin/dev/retrotv/crypto/cipher/block/algorithm/LEA.kt
package dev.retrotv.crypto.cipher.block.algorithm

import dev.retrotv.crypto.cipher.block.BlockCipher
import org.bouncycastle.crypto.engines.LEAEngine

import dev.retrotv.crypto.cipher.enums.ECipher.LEA

/**
 * LEA 암호화 알고리즘을 사용하는 블록 암호화 클래스 입니다.
 */
class LEA : BlockCipher() {
    init {
        this.engine = LEAEngine()
        this.algorithm = LEA
    }
}